### ISSUE TITLE
Add decoder-only conditional multitask model with tied vocabulary head

### ISSUE THREAD

**Issue by gkondas** [mmcdermott/EveryQuery-conditional#23]
## Summary

Add a decoder-only conditional multitask model that:

- Processes the patient history and a teacher-forced sequence of prior code-answer conditions with one causal Llama backbone.
- Produces an all-vocabulary occurrence prediction at every complete window-specification token.
- Encodes both the start and end of each window using the explicit-start contract in #24.
- Uses the model's input code-embedding table as a tied all-vocabulary output head.
- Trains directly on `MultitaskBoundaryPytorchDataset` after the conditioning-code work in #22 and explicit-window-start work in #24 are implemented.

For now, support only the base observable vocabulary. Ontology-expanded embeddings and targets are out of scope.

## Development branch

Create the implementation branch directly from the current remote `feat/multitask-boundary-sampler` branch. Do not branch from `main`, another feature branch, or a stale local/worktree branch.

```bash
git fetch origin
git switch feat/multitask-boundary-sampler
git pull --ff-only origin feat/multitask-boundary-sampler
git switch -c feat/conditional-multitask-ar
```

Before implementation, verify that the required base is an ancestor of `HEAD`:

```bash
git merge-base --is-ancestor origin/feat/multitask-boundary-sampler HEAD
```

The command must exit successfully.

## Motivation

[`ConditionalQueryARModel`](https://github.com/mmcdermott/EveryQuery-conditional/blob/feat/multitask-boundary-sampler/src/every_query/model/conditional_ar_model.py) currently processes binary code-specific queries:

```text
[patient events, C1, D1, A1, C2, D2, A2, ...]
```

The hidden state at `D_i` produces one scalar logit for `C_i`.

The multitask dataset instead provides `targets[b, i, v]`, indicating whether every vocabulary code `v` occurs inside the fully resolved window `W_i`. Under #24, each window has an independently specified start and end. The new model should predict this complete vector at each window while allowing later predictions to condition on earlier teacher-forced code-answer pairs.

The closest reference is the canonical chain and dense tied readout in [`ChainProtoEQModel`](https://github.com/mmcdermott/EQ-motor-gap/blob/5e88a35a744288b78f1d4a0801a3f7dd6806b726/eq_forks/multitask/src/every_query/model/queryarch_model.py#L564-L577), especially [`_dense_loss`](https://github.com/mmcdermott/EQ-motor-gap/blob/5e88a35a744288b78f1d4a0801a3f7dd6806b726/eq_forks/multitask/src/every_query/model/queryarch_model.py#L842-L855).

## Required autoregressive sequence

For `K` windows, construct:

```text
[patient events,
 W1, C1, A1,
 W2, C2, A2,
 ...
 W(K-1), C(K-1), A(K-1),
 WK]
```

`W_i` is one complete window-specification token containing both the start and end information from #24. At each `W_i`, ordinary token-level causal attention must provide the invariant:

```text
W_i can see:
    all patient events
    W_i itself
    all W_j, C_j, A_j for j < i

W_i cannot see:
    C_i or A_i
    later windows or conditions
```

Therefore the contextualized hidden state at `W_i` estimates:

```text
P(all vocabulary events within window W_i
  | patient, earlier teacher-forced conditions, W_i)
```

The final `W_K` prediction is conditioned on all `K - 1` supplied code-answer pairs. Do not append an unused final `C_K, A_K` pair, and do not add a separate start token.

## Model

Add a new model named:

```python
ConditionalMultitaskARModel
```

Reuse the corresponding machinery from `ConditionalQueryARModel` where practical:

- A single Hugging Face `LlamaModel` over the patient and conditional query stream.
- Plain causal attention.
- Per-row packing so query tokens start immediately after each row's final real patient event.
- Shared code embeddings for patient codes, conditioning codes, event-start codes, and event-end codes.
- A start-duration MLP and a distinct end-duration MLP.
- Event-start embeddings using the start code plus a learned `start_marker`.
- Event-end embeddings using the boundary code plus the existing learned `bound_marker`.
- Learned patient/window/code/answer token-type embeddings.
- Learned block-position embeddings for condition order.
- Existing auxiliary-embedding initialization conventions.
- Optional clinical-time RoPE.

### Window-specification embedding

Consume the explicit-start fields introduced by #24 and construct one `W_i` embedding from role-specific start and end components:

```text
start_spec =
    start_duration_mlp(q_start_duration)
    OR input_embedding[q_start_code] + start_marker

end_spec =
    end_duration_mlp(q_duration)
    OR input_embedding[q_bound_code] + bound_marker

W_i =
    start_spec
    + end_spec
    + window_token_type_embedding
    + block_position_embedding
```

The end duration is relative to the resolved start, as defined by #24. The start and end projections and markers must be distinguishable. Patient codes, conditioning codes, event-start codes, and event-end codes must all use the same live input-embedding table. Do not add a fourth token to each autoregressive block.

All query-stream tokens represent questions asked at the prediction time, so they should repeat the final real patient event's clinical-time RoPE position. A future start offset or start event is query content encoded in `W_i`; it must not advance clinical-time RoPE. Physical token order and the causal mask—not RoPE—enforce information flow.

The model should accept the #22/#24 `MultitaskBoundaryBatch` fields including:

```text
code:               (B, S)
q_start_durations:  (B, K)      # 0 means prediction-time start; sentinel for event start
q_start_codes:      (B, K)      # 0 for duration-defined starts
q_durations:        (B, K)      # end duration relative to the resolved start
q_bound_codes:      (B, K)      # 0 for duration-defined ends
q_mask:             (B, K)
condition_codes:    (B, K - 1)
condition_answers:  (B, K - 1)
targets:            (B, K, V)
time_pos_ids:       optional
```

Legacy multitask batches without explicit start fields should be interpreted as zero-duration starts at prediction time, as required by #24.

## Tied all-vocabulary readout

Gather the contextualized Llama hidden state at every window-specification position:

```text
Z_W: (B, K, H)
```

Let the live input-embedding parameter for the base vocabulary be:

```text
E: (V, H)
```

Compute:

```python
logits = Z_W @ E.T + code_bias
```

with:

```text
logits:    (B, K, V)
code_bias: (V,)
```

Requirements:

- Use the contextualized Llama hidden states at `W_i`, not the raw duration-MLP embeddings.
- The output table must be tied to the live input embedding parameter rather than copied into another parameter.
- Obtain it through `get_input_embeddings()` to preserve a future ontology-extension seam.
- Initialize the learned per-code bias to `-3.0`, following the reference.
- Do not sigmoid logits before the loss.
- Keep PAD at index 0 for tensor alignment, but exclude it from the objective.
- Fail clearly if the embedding/output width and `batch.targets.shape[-1]` differ.

## Loss

Use elementwise BCE over every real boundary and every non-PAD vocabulary code:

```python
element_loss = torch.nn.functional.binary_cross_entropy_with_logits(
    logits.float(),
    batch.targets.float(),
    reduction="none",
)
```

The loss mask should combine:

```text
batch.q_mask[:, :, None]
AND
vocab_index != PAD_INDEX
```

Normalize by the number of valid elements.

The multitask artifact has no separate censor-validity tensor. Use the materialized labels from #24 as-is—including its empty all-false windows for unresolved event starts—and do not re-resolve windows or invent censoring semantics inside the model.

## Outputs and Lightning integration

Add a multitask-specific output type exposing at least:

```text
logits:     (B, K, V)
valid_mask: broadcastable to (B, K, V)
```

Add a dedicated Lightning module for this output contract. The existing conditional Lightning module assumes scalar `(B, K)` answer logits and cannot directly consume this model's output.

The initial wrapper should:

- Log train, tuning, and held-out loss.
- Avoid accumulating every `(B, K, V)` value in an epoch-level `BinaryAUROC`, which can require substantial memory.
- Return full-vocabulary probabilities during explicit prediction.
- Save an architecture key and sufficient hyperparameters for checkpoint restoration.

Full-vocabulary macro metrics can be added separately.

## Training integration

Add a Hydra configuration connecting:

```text
MultitaskBoundaryPytorchDataset
    -> ConditionalMultitaskARModel
    -> multitask Lightning module
```

Update:

- `train.py::required_position_embeddings`.
- Checkpoint/model-loader architecture dispatch.
- Any type registrations or imports required by Hydra and prediction.

For `K` boundaries, the added query-stream length is:

```text
K + 2 * (K - 1) = 3K - 2
```

A conservative `3K` position budget is acceptable, but the model and dataset must agree on `K` before training begins.

## No-ontology MVP

For this issue:

- `ontology_dir` must be `None`.
- `V` is the manifest-aligned base vocabulary from `codes.parquet`.
- The input/output embedding table covers exactly this base vocabulary.
- A non-null ontology configuration or ontology-mode manifest must raise clearly.

## Tests

Add tests covering:

- Exact stream ordering `[W_i, C_i, A_i, W_(i+1)]`.
- Output shape `(B, K, V)`.
- Every `W_i` is supervised against `targets[:, i, :]`.
- PAD does not affect the loss.
- The tied readout uses the same parameter as the input embedding table.
- Loss gradients reach the tied embedding table.
- Changing `A_i` cannot affect the prediction at `W_i`.
- Changing `A_i` can affect later window predictions.
- Changing future conditions cannot affect earlier window predictions.
- All six start/end combinations from #24 produce valid window tokens.
- Variable patient lengths start the query stream immediately after the last real event.
- RoPE-time enabled/disabled pairing is validated.
- Explicit future starts do not advance the query-stream clinical-time RoPE position.
- Prediction-time, positive-duration, and event-defined starts all work with both end forms.
- Start-duration and end-duration projections are role-distinct.
- Event-start and event-end markers are role-distinct.
- Changing a start specification can change the corresponding `W_i` logits.
- No separate start token is inserted into the autoregressive stream.
- Legacy zero-start batches preserve prediction-time-start behavior.
- Vocabulary-width and window-count mismatches fail clearly.
- Non-null ontology configuration fails.
- A small end-to-end training run has finite loss and can save/reload a checkpoint.
- Existing single-query and conditional models continue to pass their tests.

## Acceptance criteria

- The implementation branch was created directly from the current `feat/multitask-boundary-sampler` branch.
- The model trains directly on the completed multitask sampler/dataset contract from #22 and #24.
- Every real window produces logits over the complete base vocabulary.
- Earlier materialized code-answer pairs condition later predictions without leaking the current answer into its own prediction.
- The all-vocabulary output table is tied to the input code embeddings.
- PAD is excluded from the loss.
- Prediction-time, positive-duration, and event-defined starts work with both duration- and event-defined ends.
- A Hydra config can run a train/validation smoke test and restore its checkpoint.
- The scalar AR model and other existing models remain behaviorally unchanged.
- Ontology support is explicitly rejected for this MVP.
